cmd/subctl/uint16slice.go

```diff
@@ -0,0 +1,56 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package subctl
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+type uint16Slice struct {
+	value *[]uint16
+}
+
+func (s *uint16Slice) Type() string {
+	return "uint16Slice"
```

Comment:
This appears in the help output, to match the `uintslice` output it would be better as `uint16slice`:

```
Flags:
  -h, --help                        help for prepare
      --metrics-port uint16Slice    Metrics port (default [8080 8081])
      --nat-discovery-port uint16   NAT discovery port (default 4490)
      --natt-port uint16            IPSec NAT traversal port (default 4500)
      --vxlan-port uint16           Internal VXLAN port (default 4800)
```

(I suspect that the types in help messages are rather meaningless to many users, but that’s another concern.)

Reply:
So you want the type with a lowercase ‘s’?   I made it upper case to match unitSlice as they have code that looks for “Slice” for certain functionality.

Reply:
Ah, no, this is only about the help output. The Cobra `UintSlice` produces `uintslice` somehow in the help output, but I see its `Type()` function returns `uintSlice` so please don’t change that.

Reply:
yeah I'd think `Type()` is used directly in usage output.